src/test/java/com/cag/cagbackendapi/controllers/UserControllerTest.kt

```diff
@@ -242,4 +241,83 @@ class UserControllerTest {
     }
 
+    @Test
+    fun deleteUser_validInput_200OK() {
+        val testAuthKey = "testAuthKey"
+        val userId = "123e4567-e89b-12d3-a456-426614174000"
+        val userUUID = UUID.fromString(userId)
+        val userData = UserResponseDto(userUUID, "John", "Smith", "johnjohn@aol.com", true, null)
+
+        doNothing().whenever(validationService).validateAuthKey(testAuthKey)
+        whenever(userService.deleteUser(userId)).thenReturn(userData)
+
+        userController.deleteUser(testAuthKey, userId)
+
+        verify(validationService).validateAuthKey(testAuthKey)
+        verify(userService).deleteUser(userId)
+        verifyNoMoreInteractions(validationService, userService)
+    }
+
+    @Test
+    fun deleteUser_missingUserId_400BadRequest(){
+        val testAuthKey = "testAuthKey"
+        val userId = "test user"
+
+        val badRequestException = BadRequestException(DetailedErrorMessages.INVALID_USER_ID, null)
+
+        doNothing().whenever(validationService).validateAuthKey(testAuthKey)
+        whenever(userService.deleteUser(userId)).thenThrow(badRequestException)
+
+        val actual = assertThrows<BadRequestException> {
+            userController.deleteUser(testAuthKey, userId)
+        }
+
+        assertEquals(actual.message, badRequestException.message)
+
+        verify(validationService).validateAuthKey(testAuthKey)
+        verify(userService).deleteUser(userId)
+        verifyNoMoreInteractions(validationService, userService)
+    }
+
+    @Test
+    fun deleteUser_invalidAuthKey_401Unauthorized(){
+        val testAuthKey = "testAuthKey"
+        val userId = "test user"
+
+        val unauthorizedException = UnauthorizedException(DetailedErrorMessages.MISSING_AUTH_KEY, null)
+
+        whenever(validationService.validateAuthKey(testAuthKey)).thenThrow(unauthorizedException)
+
+        val actual = assertThrows<UnauthorizedException> {
+            userController.deleteUser(testAuthKey, userId)
+        }
+
+        assertEquals(actual.message, unauthorizedException.message)
+
+        verify(validationService).validateAuthKey(testAuthKey)
+        verifyZeroInteractions(userService)
+    }
+
+
+    //REVIEW THIS TEST
+    @Test
+    fun deleteUser_invalidUserId_404UserNotFound(){
+        val testAuthKey = "testAuthKey"
+        val userId = "nonUserUUID"
+
+        val badRequestException = BadRequestException(DetailedErrorMessages.USER_NOT_FOUND, null)
+
+        doNothing().whenever(validationService).validateAuthKey(testAuthKey)
+        whenever(userService.deleteUser(userId)).thenThrow(badRequestException)
+
+        val actual = assertThrows<BadRequestException> {
+            userController.deleteUser(testAuthKey, userId)
+        }
+
+        assertEquals(actual.message, badRequestException.message)
+
+        verify(validationService).validateAuthKey(testAuthKey)
+        verify(userService).deleteUser(userId)
+        verifyNoMoreInteractions(validationService, userService)
+    }
```

Comment:
I know it's tedious, but we should also add ServiceUnavailable and InternalServerError tests for the controller, DAO, and service. It's a lot of testing but it's all copy/paste code that doesn't take long and can save us in the long run since it deters bugs.